src/main/scala/mesosphere/marathon/upgrade/DeploymentDeathRow.scala

```diff
@@ -0,0 +1,18 @@
+package mesosphere.marathon.upgrade
+
+import mesosphere.marathon.Protos.MarathonTask
+
+private[upgrade] case class DeploymentDeathRow(runningTasks: Set[MarathonTask],
+                                               sentencedToDeath: Set[MarathonTask],
+                                               scaleTo: Int) {
+
+  val tasksToBeKilled: Seq[MarathonTask] = {
+    val (sentencedAndRunning, notSentencedAndRunning) = runningTasks partition sentencedToDeath
+    val keepAliveCount = runningTasks.size - math.max(sentencedAndRunning.size, runningTasks.size - scaleTo)
```

Comment:
Wouldn't it be easier to calculate the `killCount` and use `take` instead of `dropRight` below?